core/templates/base.html

```diff
@@ -17,41 +17,80 @@
-            <button class="btn btn-success my-2 my-sm-0" type="submit">Go</button>
-        </form>
-    </nav>
-    
-    <!-- Main body -->
-    <div class="container-fluid pl-0 pr-0">
-        {% block header %}
-        {% endblock %}
-
-        {% block body %}
-        {% endblock body %}
-    </div>
+    <div class="row">
+        <div id="mainmenu" class="d-none pr-0 mr-0"></div>
+
+        <div id="maincontent" class="col-12">
 
-    <div class="container-fluid border-top justify-content-center pt-3">
-        {% if request.user.is_authenticated %}
-        <p class="text-center text-light">
-            <a class="text-secondary" href="/wiki/feed">RSS</a> | <a class="text-secondary" href="{% url "history" %}">History</a> | <a class="text-secondary" href="{% url "self" %}">Settings</a> | <a class="text-secondary" href="/logout/">Log out</a>
-        </p>
-        {% endif %}
+            <!-- Top nav -->
+            <nav class="navbar navbar-dark bg-primary">
+
+                <div>
+                    <a class="navbar-brand" href="#"><i id="mainmenutoggle" class="bi bi-list text-white"></i></a> 
+                    <a class="navbar-brand" href="{% url "wiki" %}"> Wiki</a>
+                </div>
+
+                <form class="form-inline my-2 my-lg-0" method="GET" action="{% url "search" %}">
+                    <input class="form-control mr-sm-2" type="search" placeholder="Search" name="q">
+                    <button class="btn btn-success my-2 my-sm-0" type="submit">Go</button>
+                </form>
+            </nav>
+            
+            <!-- Main body -->
+            <div class="container-fluid pl-0 pr-0">
+                {% block header %}
+                {% endblock %}
+
+                {% block body %}
+                {% endblock body %}
+            </div>
+
+            <div class="container-fluid border-top justify-content-center pt-3">
+                {% if request.user.is_authenticated %}
+                <p class="text-center text-light">
+                    <a class="text-secondary" href="/wiki/feed">RSS</a> | <a class="text-secondary" href="{% url "history" %}">History</a> | <a class="text-secondary" href="{% url "self" %}">Settings</a> | <a class="text-secondary" href="/logout/">Log out</a>
+                </p>
+                {% endif %}
+            </div>
+        </div>
     </div>
 </body>
 
 {% block extra_js %}{% endblock %}
+<script>
+    $(document).ready(function () {
+
+        $('#mainmenu').load('{{ menu_url }}', function() {
+            $('#mainmenutoggle').click(function () {
+                if ($('#mainmenu').is(":visible")) {
+                    $('#mainmenu').hide();
+                    $('#mainmenu').removeClass('col-2');
+                    $('#maincontent').removeClass('col-10');
+                    $('#maincontent').addClass('col-12');
+                    $('#maincontent').removeClass('pl-0');
```

Comment:
methinks these are 'chainable'
```suggestion
                    $('#mainmenu').removeClass('col-2').hide();
                    $('#maincontent').removeClass('col-10 pl-0').addClass('col-12');
```